test/e2e/e2e.go

```diff
@@ -16,6 +17,11 @@ import (
 	"knative.dev/serving/test"
 )
 
+var (
+	// DefaultTransport is the default http.Transport used in E2E tests
+	DefaultTransport = http.DefaultTransport.(*http.Transport)
```

Comment:
This is what I mean. This might break at some point, if underlying go impl changes types.

Reply:
I was going to use networking.Autotransport, but guess what, it is returned as http.RoundTripper and not *http.Transport as it probably should :-(

Reply:
&http.Transport{} should be enough. Let me try this.